### PR TITLE
Chg(gha): python version to matrix of 3.6-11

### PR DESCRIPTION
**Pull request by jusjusjus**
This PR fixes issue #126 .

We haven't updated the versions of numpy (and python for that matter) appropriately.  I reviewed the currently stable versions of python which is 3.8 through 11 and added these to the tests.  I then performed changes accordingly.  No code change should affect the functionality of mffpy since it's mostly adding type annotations.

### CODE REVIEW

File: .github/workflows/lint_and_test.yaml [hunk 1]
```diff
@@ -14,17 +14,20 @@ jobs:
   test_mffpy:
     # Name the job
     name: Lint and Test
+    strategy:
+      matrix:
+        python-version: ["3.6.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (jusjusjus) [inline .github/workflows/lint_and_test.yaml:19]: here we add more versions

File: mffpy/bin_files.py [hunk 2]
```diff
@@ -85,7 +85,8 @@ def scale(self) -> float:
         return self._scale
 
     def get_physical_samples(self, t0: float = 0.0,
-                             dt: float = None, block_slice: slice = None,
+                             dt: Optional[float] = None,
+                             block_slice: Optional[slice] = None,
```
Comment (jusjusjus) [inline mffpy/bin_files.py:89]: newer mypy switched default for implicit optional

File: mffpy/tests/test_devices.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def test_devices(device):
     locs = np.array([
         np.array([props['x'], props['y'], props['z']])
         for i, (_, props) in enumerate(coords.sensors.items())
-    ], dtype=np.float)
+    ], dtype=np.float32)
```
Comment (jusjusjus) [inline mffpy/tests/test_devices.py:49]: np.float removed in numpy v1.20

**Review by damian5710 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**